include/libtorrent/config.hpp

```diff
@@ -173,7 +173,7 @@ POSSIBILITY OF SUCH DAMAGE.
 // FreeBSD has a reasonable iconv signature
 // unless we're on glibc
 #ifndef __GLIBC__
-# define TORRENT_ICONV_ARG(x) (x)
+# define TORRENT_ICONV_ARG(x) (const_cast<char**>(x))
```

Comment:
If this patch works, the comment should be removed as well, in fact, this ``#define`` should be removed. The default for this macro is already to perform the ``const_cast``.